pkg/validation/limits.go

```diff
@@ -0,0 +1,252 @@
+package validation
+
+import (
+	"encoding/json"
+	"flag"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/model"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	bytesInMB = 1048576
+)
+
+// Limits describe all the limits for users; can be used to describe global default
+// limits via flags, or per-user limits via yaml config.
+// NOTE: we use custom `model.Duration` instead of standard `time.Duration` because,
+// to support user-friendly duration format (e.g: "1h30m45s") in JSON value.
+type Limits struct {
+	// Distributor enforced limits.
+	IngestionRateMB        float64 `yaml:"ingestion_rate_mb" json:"ingestion_rate_mb"`
+	IngestionBurstSizeMB   float64 `yaml:"ingestion_burst_size_mb" json:"ingestion_burst_size_mb"`
+	MaxLabelNameLength     int     `yaml:"max_label_name_length" json:"max_label_name_length"`
+	MaxLabelValueLength    int     `yaml:"max_label_value_length" json:"max_label_value_length"`
+	MaxLabelNamesPerSeries int     `yaml:"max_label_names_per_series" json:"max_label_names_per_series"`
+
+	// Ingester enforced limits.
+	MaxLocalSeriesPerUser  int `yaml:"max_series_per_user" json:"max_series_per_user"`
+	MaxGlobalSeriesPerUser int `yaml:"max_global_series_per_user" json:"max_global_series_per_user"`
```

Comment:
I have spent a lot of time explaining the difference to GEx customers between global and local per instance limits. I wonder if we really need them both. What is the use case?

At least the yaml should be changing from `max_series_per_user` -> `max_local_series_per_user`

Reply:
The use case is to set a max limit per ingester regardless of how many ingester exists in the ring. e.g small cluster.